session.go

```diff
@@ -1,97 +1,77 @@
 package bond
 
 import (
+	"database/sql"
+	"fmt"
 	"reflect"
 	"sync"
 
-	"upper.io/db"
+	"upper.io/db.v2"
+	"upper.io/db.v2/lib"
 )
 
 type Session interface {
-	db.Tx
+	lib.SQLDatabase
+
 	Store(interface{}) Store
 	Find(...interface{}) db.Result
 	Save(Model) error
 	Delete(Model) error
-	NewTransaction() (Session, error)
-	ContinueTransaction() (Session, error, bool)
+
+	BondTx(func(tx Session) error) error
```

Comment:
We can't use `Tx()` because this struct uses the (temporary) `SQLDatabase` interface, which already has its own `Tx()` and the names collide, we need something to differentiate a bond transaction (which receives a bond session) from a upper transaction. `WithTx()` could be fine, but `Tx()` will still be around and the difference between `WithTx` and `Tx()`.

We can also take over the name `Tx()` and only provide a bond transaction by doing something like `Database lib.SQLDatabase`), we'll just have to add a bunch of methods on Session to be able to do things like `sess.Collection()` instead of `sess.Database.Collection()`.